boundary/collect.py

```diff
@@ -303,36 +331,38 @@ def each_call_stmt(self, node):
                 if isinstance(stmt, gcc.GimpleCall):
                     yield stmt
 
-    def process_passes(self, p, _):
+    def collect_info(self, p, _):
+        """Collect information about the current source file"""
         if p.name != '*free_lang_data':
             return
 
-        self.collect_edges()
         self.collect_fn()
-        self.collect_callbacks()
+        self.collect_callback()
         self.collect_struct()
+        self.collect_edge()
         self.collect_var()
 
-        collect = {
-            "fn": self.fn_properties,
-            "var": self.var_properties,
-            "edge": self.edge_properties,
-            "callback": self.callback_properties,
-            "interface": self.interface_properties,
-            "struct": self.struct_properties
+        collection = {
+            'fn': self.fn_prop,
+            'var': self.var_prop,
+            'edge': self.edge_prop,
+            'callback': self.cb_prop,
+            'interface': self.intf_prop,
+            'struct': self.struct_prop
         }
+
         with open(gcc.get_main_input_filename() + '.boundary', 'w') as f:
-            json.dump(collect, f, indent=4)
+            json.dump(collection, f, indent=4)
 
     def register_cbs(self):
-        gcc.register_callback(gcc.PLUGIN_PASS_EXECUTION, self.process_passes)
+        """Register GCC Python Plugin callback"""
+        gcc.register_callback(gcc.PLUGIN_PASS_EXECUTION, self.collect_info)
 
 
 if __name__ == '__main__':
     import gcc
 
-    tmpdir = gcc.argument_dict['tmpdir']
-    modpath = gcc.argument_dict['modpath']
```

Comment:
We can remove

`-fplugin-arg-python-modpath=$(plugsched_modpath)`

from the Makefile now